generate_openj9_scc.sh

```diff
@@ -0,0 +1,223 @@
+
+    # Removing eclipse
+    rm -rf "$HOME"/eclipse-home
+
+}
+
+
+run_tomcat_and_stop() {
+    "$HOME"/tomcat/apache-tomcat-9.0.35/bin/startup.sh && sleep 5 && "$HOME"/tomcat/apache-tomcat-9.0.35/bin/shutdown.sh && sleep 2
+}
+
+download_tomcat() {
+
+    # Creating a temporary directory for tomcat download
+    mkdir -p "$HOME"/tomcat/
+
+    # Downloading tomcat
+    wget --directory-prefix=/tmp https://downloads.apache.org/tomcat/tomcat-9/v9.0.35/bin/apache-tomcat-9.0.35.tar.gz
+
+    # Extracting tomcat 
+    tar -xvzf /tmp/apache-tomcat-9.0.35.tar.gz -C "$HOME"/tomcat/
+
+    # Removing the tar file
+    rm -rf /tmp/apache-tomcat-9.0.35.tar.gz
+
+}
+
+dry_run_tomcat() {
+
+    # Creating base layer first instead of running tomcat server to generate SCC directly
+    java -Xshareclasses:name=default_scc,cacheDir=/opt/java/.scc,bootClassesOnly,nonFatal,createLayer -Xscmx$SCC_SIZE -version
+
+    # Pointing cache for tomcat runs
+    export OPENJ9_JAVA_OPTIONS="-Xshareclasses:name=default_scc,cacheDir=/opt/java/.scc,bootClassesOnly,nonFatal"
+
+    for ((i=0; i<SCC_GEN_RUNS_COUNT; i++))
+    do
+        run_tomcat_and_stop
+    done
+
+    FULL=$( (java -Xshareclasses:name=default_scc,cacheDir=/opt/java/.scc,printallStats || true) 2>&1 | awk '/^Cache is [0-9.]*% .*full/ {print substr($3, 1, length($3)-1)}')
+
+    java -Xshareclasses:name=default_scc,cacheDir=/opt/java/.scc,destroy || true
+
+    SCC_SIZE="${SCC_SIZE:0:-1}"
+
+    SCC_SIZE=$(awk "BEGIN {print int($SCC_SIZE * $FULL / 100.0)}")
+
+    [ "$SCC_SIZE" -eq 0 ] && SCC_SIZE=1
+
+    SCC_SIZE="${SCC_SIZE}m"
+
+    # Re-generating cache with new size
+    java -Xshareclasses:name=default_scc,cacheDir=/opt/java/.scc,bootClassesOnly,nonFatal,createLayer -Xscmx$SCC_SIZE -version
+
+    unset OPENJ9_JAVA_OPTIONS
+
+}
+
+generate_tomcat_cache() {
+
+    # Pointing cache for tomcat runs
+    export OPENJ9_JAVA_OPTIONS="-Xshareclasses:name=default_scc,cacheDir=/opt/java/.scc,bootClassesOnly,nonFatal"
+
+    # Generating the SCC
+    for ((i=0; i<SCC_GEN_RUNS_COUNT; i++))
+    do
+        run_tomcat_and_stop
+    done
+
+    # Checking the cache level
+    FULL=$( (java -Xshareclasses:name=default_scc,cacheDir=/opt/java/.scc,printallStats || true) 2>&1 | awk '/^Cache is [0-9.]*% .*full/ {print substr($3, 1, length($3)-1)}')
+
+    echo "SCC layer is $FULL% full."
+
+    # Removing tomcat 
+    rm -rf "$HOME"/tomcat
+
+}
+
+init_tomcat_cache_gen() {
+
+    download_tomcat
+    dry_run_tomcat
+    generate_tomcat_cache
+
+}
+
+remove_packages() {
+    apt --purge autoremove gtk+3.0 xvfb
+}
+
+init_eclipse_cache_gen() {
+    download_eclipse
+    install_packages
+    dry_run_eclipse
+    generate_eclipse_cache
+    remove_packages
+}
+
+if [ -f /etc/os-release ]; then
+    # shellcheck disable=SC1091
+    source /etc/os-release
+    OS=$ID
+fi
+
+if [ "$OS" == "ubuntu" ]  || [ "$OS" = "debian" ]; then
+    init_eclipse_cache_gen
+else
+    init_tomcat_cache_gen
```

Comment:
`init_tomcat_cache_gen` should be called for all OSes.

Reply:
Updated to run tomcat on all platforms. thanks for the review. Earlier was in an impression that tomcat should be run on the platforms where eclipse runs are unavailable.